### ISSUE TITLE
Shift and Ctrl not working as macros

### ISSUE THREAD

**Issue by SubstanceSix** [Ryochan7/DS4Windows#1081]
Trying to use macros that I set a long time ago for a game, they used to work but trying the game after a few months and a few updates of DS4W they now don't work. It seems that it cannot use Ctrl and Shift as macro, the letter keys do work though.

**Comment by mika-n** [comment]
Just tested the following macro sequence in the latest V2.0.7 version and it worked just fine. The "A" letter was outputted as capital-A in Notepad and "B" letter as lowercase letter.

![DS4Win_ShiftMacroOutputTest](https://user-images.githubusercontent.com/35538525/73338122-92893700-427e-11ea-99b1-bab4ffc69690.png)

Try re-record the macro and if it still doesn't work then please provide the macro sequence definition from Actions.xml config file (the file is in DS4Windows profile folder, or actually one path above the default profile folder).

**Comment by Ryusennin** [comment]
I'm using a similar macro (Shift+Alt+S) mapped to the Share button to take screenshots with Radeon Software. Sometimes DS4Windows will not register the tap, unless I tap (press/release) the Share button very fast, which can be tedious especially if I feel a bit tired.

So your problem might actually be related to tapping delay, which in my experience is too short since several releases (I seem to remember the tapping delay was changed some time ago). Some buttons like Share and Options are not ergonomic at all, and you will likely press them much longer than a few milliseconds.

**Comment by Ryochan7** [comment]
I guess this can been as a side effect of the app getting snappier with the 2.x releases. The key presses should work just fine; they work on my end. The issue is likely due to the key press and key releases happening faster than other applications can process. You will likely need some form of a wait in the macro somewhere to give other apps time to detect the initial key presses.

**Comment by mika-n** [comment]
And remember that if "Synchronized run" option is NOT checked and you have a macro feeding something like this "ABCD" and the macro execution takes few milliseconds to complete then if you press the trigger key twice very fast then the second trigger may be launched before the first trigger completed. The end result may be something like "ABACDBCD" (ie two macro executions of the same macro sequence running interleaved). 

The "synchronized run" option is very important in cases where you two macros bind to the same trigger. One with "run on trigger down" (default behaviour) and other macro with "run on trigger release". If synchronized run option is not set for these macro actions then by the time you release the trigger key then "run on release" macro is fired away even when the "key down" macro is still running the last few steps.

**Comment by SubstanceSix** [comment]
Thank you for the responses, it was actually the delay between the key registers option that I had unchecked, probably due to what Ryochan7 suggested. Thank you for the additional tips as well. I thought I have tested this out, but anyway, what riddles me is that when I tried macros with key/symbol letters (ex. "c"+"/" ) it worked fine, and with Shift and Ctrl it registered only the letter key part. Perhaps Windows induce some tiny delay waiting for a Windows shortcut or something of that nature with the press of these keys.
